protocol/v2/ssv/runner/proposer.go

```diff
@@ -189,9 +193,15 @@ func (r *ProposerRunner) ProcessConsensus(ctx context.Context, logger *zap.Logge
 		}
 	}
 
+	duty := r.BaseRunner.State.StartingDuty.(*spectypes.ValidatorDuty)
+	if r.doppelgangerHandler.ValidatorStatus(duty.ValidatorIndex) == doppelganger.SigningDisabled {
+		logger.Warn("Doppelganger check in progress, signing not permitted", fields.ValidatorIndex(duty.ValidatorIndex))
+		return nil
+	}
```

Comment:
why is this needed if you do it in `executeDuty` already?

Reply:
Good point! `executeDuty` is part of the pre-consensus phase, where we prevent signing the partial RANDAO signature if the validator is DG-unsafe. However, if we still reach ProcessConsensus, this check acts as an additional safeguard to ensure we don't sign the block in case the validator's status changed in the meantime.
Does this make sense?

Reply:
Yeah if we're able to reach it even if we didn't execute the duty? As if there was pre-consensus from the rest of the committee on the randao?